### PR TITLE
Abridged commented version of the damage PR, PART TWO (DNM) 

### PR DESCRIPTION
**Pull request by phil235**
for reviewing #20793 only.

### CODE REVIEW

File: code/game/objects/structures/fireaxe.dm [hunk 1]
```diff
@@ -6,20 +6,38 @@
...
 		toggle_lock(user)
-		return
-	if(open || health <= 0)
+	else if(istype(I, /obj/item/stack/sheet/glass) && broken)
+		var/obj/item/stack/sheet/glass/G = I
+		if(G.get_amount() < 2)
+			user << "<span class='warning'>You need two glass sheets to fix [src]!</span>"
+			return
+		user << "<span class='notice'>You start fixing [src]...</span>"
+		if(do_after(user, 20, target = src))
```
Comment (ChangelingRain) [inline code/game/objects/structures/fireaxe.dm]: `if(do_after(user, 20, target = src) && G.use(2))` so that it fails if you use the glass on something else and take it down to one sheet

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm [hunk 2]
```diff
@@ -278,8 +277,8 @@
 	else
 		M.apply_damage(0.6*reac_volume, BRUTE)
 
-/datum/reagent/blob/explosive_lattice/damage_reaction(obj/structure/blob/B, original_health, damage, damage_type, cause)
-	if(isnull(cause))
+/datum/reagent/blob/explosive_lattice/damage_reaction(obj/structure/blob/B, damage, damage_type, damage_flag)
+	if(damage_flag != "melee" || damage_flag != "bullet" || damage_flag != "laser")
```
Comment (ChangelingRain) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: These lines could be simplified/made proper, now that there's an actual way to tell if a bomb is hitting it.

    if(damage_flag == "bomb")
    	return 0
    else if(damage_flag != "melee" || damage_flag != "bullet" || damage_flag != "laser")
    	return damage * 1.5
    return ..()

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm [hunk 3]
```diff
@@ -236,8 +235,8 @@
 	if(M)
 		M.apply_damage(0.6*reac_volume, OXY)
 
-/datum/reagent/blob/energized_jelly/damage_reaction(obj/structure/blob/B, original_health, damage, damage_type, cause)
-	if(!isnull(cause) && original_health - damage <= 0 && prob(10))
+/datum/reagent/blob/energized_jelly/damage_reaction(obj/structure/blob/B, damage, damage_type, damage_flag)
+	if((damage_flag == "melee" || damage_flag == "bullet" || damage_flag == "laser") && B.obj_integrity - damage <= 0 && prob(10))
 		spark_system.set_up(rand(2, 4), 0, B)
 		spark_system.start()
 	return ..()
```
Comment (ChangelingRain) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: The B.take_damage() line for the emp_reaction for this chemical should use the "energy" armor flag.

**Review by ChangelingRain — COMMENTED** [review]
this is what I really wanted to review

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm [hunk 4]
```diff
@@ -289,10 +289,10 @@
 	feedback_add_details("slime_cores_used","[type]")
 	var/turf/TU = get_turf(holder.my_atom)
 	TU.visible_message("<span class='danger'>The slime extract begins to vibrate adorably!</span>")
-	addtimer(src, "burn", 50, FALSE, holder)
+	addtimer(src, "slime_burn", 50, FALSE, holder)
 
 
-/datum/chemical_reaction/slimefire/proc/burn(datum/reagents/holder)
+/datum/chemical_reaction/slimefire/proc/slime_burn(datum/reagents/holder)
```
Comment (phil235) [inline code/modules/reagents/chemistry/recipes/slime_extracts.dm]: changed the name so you don't confuse the proc with obj/proc/burn(),  it also makes searches easier.

File: code/modules/surgery/bodyparts/bodyparts.dm [hunk 5]
```diff
@@ -104,7 +104,7 @@
 //Applies brute and burn damage to the organ. Returns 1 if the damage-icon states changed at all.
 //Damage will not exceed max_damage using this proc
 //Cannot apply negative damage
-/obj/item/bodypart/proc/take_damage(brute, burn, updating_health = 1)
+/obj/item/bodypart/proc/receive_damage(brute, burn, updating_health = 1)
```
Comment (phil235) [inline code/modules/surgery/bodyparts/bodyparts.dm]: changed the proc name since it doesn't do the same thing as /obj/proc/take_damage(). Maybe in the future we'll have the bodypart actually use a version of take_damage(), but for now we have two procs doing different things, so two names to avoid confusion.

File: code/game/objects/structures/watercloset.dm [hunk 6]
```diff
@@ -274,6 +274,7 @@
 
 	if(istype(O,/obj/item))
 		var/obj/item/I = O
+		I.acid_level = 0
```
Comment (phil235) [inline code/game/objects/structures/watercloset.dm]: the water washes the acid away.

File: code/modules/mob/living/carbon/carbon.dm [hunk 7]
```diff
@@ -501,8 +501,6 @@
 	update_stat()
 	if(((maxHealth - total_burn) < HEALTH_THRESHOLD_DEAD) && stat == DEAD )
 		become_husk()
-		if(on_fire)
-			shred_clothing()
```
Comment (phil235) [inline code/modules/mob/living/carbon/carbon.dm]: Your clothes don't automatically shred when you become a husk (was especially weird for spaced bodies).

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 8]
```diff
@@ -18,8 +18,8 @@
 	return (armorval/max(organnum, 1))
 
 
-/mob/living/carbon/human/proc/checkarmor(obj/item/bodypart/def_zone, type)
-	if(!type)
+/mob/living/carbon/human/proc/checkarmor(obj/item/bodypart/def_zone, d_type)
+	if(!d_type)
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/human_defense.dm]: changed the argument name, we don't want args sharing names with actual variables, just  to be safe.

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 9]
```diff
@@ -190,7 +191,8 @@
 /mob/living/carbon/human/attack_paw(mob/living/carbon/monkey/M)
 	var/dam_zone = pick("chest", "l_hand", "r_hand", "l_leg", "r_leg")
 	var/obj/item/bodypart/affecting = get_bodypart(ran_zone(dam_zone))
-
+	if(!affecting)
+		affecting = get_bodypart("chest")
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/human_defense.dm]: so we don't attack a missing limb. If we can't find the limb, we default to chest.

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 10]
```diff
@@ -323,49 +340,50 @@
...
 		return
+	var/b_loss = 0
+	var/f_loss = 0
+	var/bomb_armor = getarmor(null, "bomb")
 
 	switch (severity)
 		if (1)
-			b_loss += 500
-			if (prob(bomb_armor))
-				shred_clothing(1,150)
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/human_defense.dm]: we don't need to shred clothes here, ex_act() calls ..() all the way down to atom/ex_act() which calls contents_explosion() which calls ex_act() on all the clothes.

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 11]
```diff
@@ -323,49 +340,50 @@
...
-			if (prob(bomb_armor))
-				b_loss = b_loss/1.5
-				f_loss = f_loss/1.5
-				shred_clothing(1,25)
-			else
-				shred_clothing(1,50)
+			b_loss = 60
+			f_loss = 60
+			if(bomb_armor)
+				b_loss = 30*(2 - round(bomb_armor*0.01, 0.05))
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/human_defense.dm]: between 30 and 60 depending on your armor.

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 12]
```diff
@@ -323,49 +340,50 @@
...
+
 			if (!istype(ears, /obj/item/clothing/ears/earmuffs))
 				adjustEarDamage(15,60)
 			if (prob(50))
 				Paralyse(8)
 
 	take_overall_damage(b_loss,f_loss)
+	if(bomb_armor == 100) //full bomb armor set, we don't call contents_explosion
+		flash_act()
+		return
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/human_defense.dm]: This is to avoid contents_explosion() calling ex_act on our clothes. I guess it's a bit of a nerf, since now wearing some bomb armor but not a full set still expose all your clothes to be shredded.

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 13]
```diff
@@ -200,100 +200,6 @@
...
-
-
-//Cycles through all clothing slots and tests them for destruction
-/mob/living/carbon/proc/shred_clothing(bomb,shock)
-	if(back)
-		back.shred(bomb,shock-20,src)
-	if(head)
-		head.shred(bomb,shock,src)
-	if(wear_mask)
-		wear_mask.shred(bomb,shock,src)
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/inventory.dm]: I had to remove shred_clothing()  since ex_act()'s ..() already calls contents_explosion. So it was either this or have human/ex_act() not call ..() which would be inconsistent with every other ex_act(). We lose some accuracy in the process sadly, a suit with some bomb armor doesn't protect your jumpsuit from bomb effect anymore, unless you have bomb suit + bomb hood on you. This is a nerf.

File: code/modules/paperwork/filingcabinet.dm [hunk 14]
```diff
@@ -53,9 +54,11 @@
 		icon_state = initial(icon_state)
 		updateUsrDialog()
 	else if(istype(P, /obj/item/weapon/wrench))
+		user << "<span class='notice'>You begin to [anchored ? "wrench" : "unwrench"] [src].</span>"
 		playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
-		anchored = !anchored
-		user << "<span class='notice'>You [anchored ? "wrench" : "unwrench"] [src].</span>"
+		if(do_after(user, 20, target = src))
+			user << "<span class='notice'>You successfully [anchored ? "wrench" : "unwrench"] [src].</span>"
+			anchored = !anchored
```
Comment (phil235) [inline code/modules/paperwork/filingcabinet.dm]: wrenching a filingcabinet is no longer instantaneous.

File: code/modules/power/apc.dm [hunk 15]
```diff
@@ -183,7 +187,6 @@
 /obj/machinery/power/apc/examine(mob/user)
 	..()
 	if(stat & BROKEN)
-		user << "Looks broken."
```
Comment (phil235) [inline code/modules/power/apc.dm]: already shown in machinery/examine()

File: code/modules/power/apc.dm [hunk 16]
```diff
@@ -555,13 +558,21 @@
...
 
-/obj/machinery/power/apc/take_damage(damage, damage_type = BRUTE, sound_effect = 1)
-	..()
-	if((malfhack || (stat & BROKEN)) && !opened)
-		if(damage < 10)
-			return
-		health -= damage
-		if(health <= 0)
+/obj/machinery/power/apc/run_obj_armor(damage_amount, damage_type, damage_flag = 0, attack_dir)
+	if(damage_flag == "melee" && damage_amount < 15 && (!(stat & BROKEN) || malfai))
```
Comment (phil235) [inline code/modules/power/apc.dm]: The apc ignores hit below 15, so an assistant with an extinguisher can't break all apcs for fun. But once the apc is broken open the threshold isn't applied so that dropping the cover to fix the apc isn't tedious.

File: code/modules/power/apc.dm [hunk 17]
```diff
@@ -602,28 +613,9 @@
...
-			update_icon()
-			visible_message("<span class='danger'>The [src.name]'s wires are shredded!</span>")
-	else if(opened == 1)
-		if(cell)
-			cell.loc = user.loc
-			cell.updateicon()
-			cell = null
-			visible_message("<span class='danger'>The [src.name]'s power cell flies off!</span>")
-			charging = 0
-			update_icon()
```
Comment (phil235) [inline code/modules/power/apc.dm]: keeping this special behavior would conflict with the standard damage behavior.

File: code/modules/power/apc.dm [hunk 18]
```diff
@@ -555,13 +558,21 @@
...
+
+/obj/machinery/power/apc/obj_break(damage_flag)
+	if(!(flags & NODECONSTRUCT))
+		set_broken()
+
+/obj/machinery/power/apc/deconstruct(disassembled = TRUE)
+	if(!(flags & NODECONSTRUCT))
+		if(!(stat & BROKEN))
+			set_broken()
+		if(opened != 2)
```
Comment (phil235) [inline code/modules/power/apc.dm]: apc is one of the very few obj to not be qdel()'d by its deconstruct().

File: code/modules/power/cable.dm [hunk 19]
```diff
@@ -97,9 +97,16 @@ By design, d1 is the smallest direction and d2 is the highest
 	if(invisibility != INVISIBILITY_MAXIMUM)
 		qdel(src)
 
-/obj/structure/cable/Deconstruct()
-	var/turf/T = loc
-	stored.loc = T
+/obj/structure/cable/deconstruct(disassembled = TRUE)
+	if(!(flags & NODECONSTRUCT))
```
Comment (phil235) [inline code/modules/power/cable.dm]: all these NODECONSTRUCT check in deconstruct() and obj_break() are necessary so if someone makes holo cables they don't drop real cable coils.

File: code/modules/power/lighting.dm [hunk 20]
```diff
@@ -16,34 +16,38 @@
...
 /obj/item/wallframe/light_fixture/small
 	name = "small light fixture frame"
 	icon_state = "bulb-construct-item"
-	result_path = /obj/machinery/light_construct/small
+	result_path = /obj/structure/light_construct/small
 	materials = list(MAT_METAL=MINERAL_MATERIAL_AMOUNT)
 
 
-/obj/machinery/light_construct
+/obj/structure/light_construct
```
Comment (phil235) [inline code/modules/power/lighting.dm]: it's a frame, not a machine.

File: code/modules/power/lighting.dm [hunk 21]
```diff
@@ -475,26 +493,13 @@
...
-	L.loc = loc
-
-	status = LIGHT_EMPTY
-	update()
 
 // break the light and make sparks if was on
 
-/obj/machinery/light/proc/broken(skip_sound_and_sparks = 0)
-	if(status == LIGHT_EMPTY)
+/obj/machinery/light/proc/break_light_tube(skip_sound_and_sparks = 0)
```
Comment (phil235) [inline code/modules/power/lighting.dm]: proc name change to be more explicit and not share a name with many other procs.

File: code/modules/power/singularity/containment_field.dm [hunk 22]
```diff
@@ -26,11 +26,20 @@
 		shock(user)
 		return 1
 
+/obj/machinery/field/containment/attackby(obj/item/W, mob/user, params)
+	shock(user)
+	return 1
```
Comment (phil235) [inline code/modules/power/singularity/containment_field.dm]: "Urist has attacked the containment field with a toolbox"

File: code/modules/power/singularity/particle_accelerator/particle_accelerator.dm [hunk 23]
```diff
@@ -26,6 +26,10 @@
 	icon_state = "none"
 	anchored = 0
 	density = 1
+	obj_integrity = 500
+	max_integrity = 500
```
Comment (phil235) [inline code/modules/power/singularity/particle_accelerator/particle_accelerator.dm]: Some decent hp and melee armor for balance purposes, don't want an assistant to destroy one of the unique engine parts too easily.

File: code/modules/power/singularity/singularity.dm [hunk 24]
```diff
@@ -60,6 +60,19 @@
...
+
+/obj/singularity/attack_alien(mob/user)
+	consume(user)
+
+/obj/singularity/attack_animal(mob/user)
+	consume(user)
+
+/obj/singularity/attackby(obj/item/weapon/W, mob/user, params)
+	consume(user)
+	return 1
```
Comment (phil235) [inline code/modules/power/singularity/singularity.dm]: "Urist has attacked the singularity with a toolbox"
Believe it or not it actually works, I tested it, lol.

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm (hunk 3 above)
Comment (phil235) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: willl fix (line 249)

File: code/modules/research/server.dm [hunk 25]
```diff
@@ -3,7 +3,7 @@
 	icon = 'icons/obj/machines/research.dmi'
 	icon_state = "server"
 	var/datum/research/files
-	var/health = 100
+	var/heat_health = 100
```
Comment (phil235) [inline code/modules/research/server.dm]: name change to avoid confusion.

File: code/modules/vehicles/vehicle.dm [hunk 26]
```diff
@@ -144,3 +147,28 @@
...
+		var/healthpercent = (obj_integrity/max_integrity) * 100
+		switch(healthpercent)
+			if(100 to INFINITY)
+				user <<  "It seems pristine and undamaged."
+			if(50 to 100)
+				user <<  "It looks slightly damaged."
+			if(25 to 50)
+				user <<  "It appears heavily damaged."
+			if(0 to 25)
+				user <<  "<span class='warning'>It's falling apart!</span>"
```
Comment (phil235) [inline code/modules/vehicles/vehicle.dm]: I had to put some nearly copypasta for the examine() of vehicle/machinery/structure, because I can't have it at /obj/ level. It would appear on decals, smoke effect, and items which we don't want.

File: code/modules/reagents/chemistry/reagents/blob_reagents.dm (hunk 2 above)
Comment (phil235) [inline code/modules/reagents/chemistry/reagents/blob_reagents.dm]: Alright, I'll use your code.
